File: src/Oisys.Web/wwwroot/js/models/cash-voucher.js
(function (module) {
    module.factory("CashVoucher", function () {
        function CashVoucher(id, voucherNumber, date, amount) {
            this.id = id || 0;
            this.voucherNumber = voucherNumber || 0;

            this.date = date || new Date();
            this.amount = amount || 0;
        };

        return CashVoucher;
    });

})(angular.module("oisys-app"));

File: src/Oisys.Web/Models/CashVoucher.cs
﻿namespace Oisys.Web.Models
{
    using System;

    /// <summary>
    /// The cash voucher model
    /// </summary>
    public class CashVoucher : ModelBase
    {
        /// <summary>
        /// Gets or sets the voucher number
        /// </summary>
        public int VoucherNumber { get; set; }

        /// <summary>
        /// Gets or sets the voucher date
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Gets or sets pay to
        /// </summary>
        public string PayTo { get; set; }

        /// <summary>
        /// Gets or sets description
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Gets or sets amount
        /// </summary>
        public decimal Amount { get; set; }

        /// <summary>
        /// Gets or sets released by
        /// </summary>
        public string ReleasedBy { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether cash voucher is deleted.
        /// </summary>
        public bool IsDeleted { get; set; }

        /// <summary>
        /// Gets or sets category
        /// </summary>
        public string Category { get; set; }
    }
}